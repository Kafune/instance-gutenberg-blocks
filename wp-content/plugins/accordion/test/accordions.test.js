

import Accordion from "./../src/components/accordion";
import Accordions from "./../src/components/accordions";
import React from "react";
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';


describe("Accordions functions", () => {
	const accordions = [
		{
			heading: "Title that can be expanded",
			description:
				"By creating an Account on our service, you agree to subscribe to newsletters, marketing or promotional materials and other information we may send. However, you may opt out of receiving any, or all, of these communications from us by following the unsubscribe link or instructions provided in any email we send.",
		},
		{
			heading:
				"This is an example title that has enough text to spread over two lines ",
			description:
				"By creating an Account on our service, you agree to subscribe to newsletters, marketing or promotional materials and other information we may send. However, you may opt out of receiving any, or all, of these communications from us by following the unsubscribe link or instructions provided in any email we send.",
		},
		{
			heading: "Another title that can be expanded",
			description:
				"By creating an Account on our service, you agree to subscribe to newsletters, marketing or promotional materials and other information we may send. However, you may opt out of receiving any, or all, of these communications from us by following the unsubscribe link or instructions provided in any email we send.",
		},
		{
			heading: "This is an item that has been expanded",
			description:
				"By creating an Account on our service, you agree to subscribe to newsletters, marketing or promotional materials and other information we may send. However, you may opt out of receiving any, or all, of these communications from us by following the unsubscribe link or instructions provided in any email we send.",
		},
	]

	const testAccordion = {
		heading: "Title that can be expanded",
		description:
			"By creating an Account on our service, you agree to subscribe to newsletters, marketing or promotional materials and other information we may send. However, you may opt out of receiving any, or all, of these communications from us by following the unsubscribe link or instructions provided in any email we send.",
	};

	beforeEach(() => {});

	test("Snapshot test", () => {
		const wrapper = shallow(
			<Accordion
				heading={testAccordion.heading}
				description={testAccordion.description}
				changeProp={(property) => changeProp({ property }, i)}
			/>
		);

		// const accordion = jest.mock('./../src/components/accordion',() => 'Accordion')

		// expect(wrapper.find(Accordion).prop('heading')).toBe("Title that can be expanded");
		// const container = wrapper.find(Accordion).length
		// console.log(toJson(wrapper).find(Accordion).length)
		console.log(wrapper.find('heading'))

	});

	test("Mocking Accordion", () => {
		jest.mock('./../src/components/accordions', () => 'Accordions')
		// jest.doMock('./../src/components/accordion', () => {
		// 	const component = () => <div/>
		// 	return component;
		// })

		// const test = require('./../src/components/accordions').default;
		console.log(jest.mock('./../src/components/accordions', () => 'Accordions')
		)
	})
});
