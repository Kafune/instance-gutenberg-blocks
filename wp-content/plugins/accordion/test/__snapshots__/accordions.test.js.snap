// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordions functions Shallow render single accordion 1`] = `
<div
  class="accordion"
>
  <div
    class="accordion-heading"
  >
    <h4
      aria-activedescendant={null}
      aria-label="Title..."
      aria-multiline={true}
      aria-owns={null}
      className="block-editor-rich-text__editable rich-text"
      contentEditable={true}
      onKeyDown={[Function]}
      role="textbox"
      suppressContentEditableWarning={true}
    />
    <span
      class="accordion-icon"
    >
      -
    </span>
  </div>
  <div
    class="accordion-description accordion-open"
  >
    <p
      aria-activedescendant={null}
      aria-label="Enter your description..."
      aria-multiline={true}
      aria-owns={null}
      className="block-editor-rich-text__editable rich-text"
      contentEditable={true}
      onKeyDown={[Function]}
      role="textbox"
      suppressContentEditableWarning={true}
    />
  </div>
</div>
`;

exports[`Accordions functions Shallow render single accordion 2`] = `
<div
  class="accordion"
>
  <div
    class="accordion-heading"
  >
    <h4
      aria-activedescendant={null}
      aria-label="Title..."
      aria-multiline={true}
      aria-owns={null}
      className="block-editor-rich-text__editable rich-text"
      contentEditable={true}
      onKeyDown={[Function]}
      role="textbox"
      suppressContentEditableWarning={true}
    />
    <span
      class="accordion-icon"
    >
      -
    </span>
  </div>
  <div
    class="accordion-description accordion-open"
  >
    <p
      aria-activedescendant={null}
      aria-label="Enter your description..."
      aria-multiline={true}
      aria-owns={null}
      className="block-editor-rich-text__editable rich-text"
      contentEditable={true}
      onKeyDown={[Function]}
      role="textbox"
      suppressContentEditableWarning={true}
    />
  </div>
</div>
`;
